docs: streaming metadata, dynamic pages and generateMetadata (#80453)
vercel/next.js: docs/01-app/05-api-reference/04-functions/generate-metadata.mdx

Metadata returned by `generateMetadata` is streamed to the client. This allows Next.js to inject metadata into the HTML as soon as it's resolved.

> Streamed metadata may end up in the `<body>` tag. Since metadata mainly targets bots and crawlers, Next.js streams metadata for bots that can execute JavaScript and inspect the full DOM (e.g. `Googlebot`). We have verified that these bots, interpret the metadata correctly.

Reviewer: May or is? 

```suggestion
Streamed metadata is appended to the `<body>` tag. Since metadata mainly targets bots and crawlers, Next.js streams metadata for bots that can execute JavaScript and inspect the full DOM (e.g. `Googlebot`). We have verified that these bots interpret the metadata correctly.
```

> For **HTML-limited** bots that can’t run JavaScript (e.g. `Twitterbot`), metadata continues to block page rendering and is placed in the `<head>` tag.

Reviewer: 👍🏼 Good distinction
